kernel/init.c

```diff
@@ -403,6 +406,84 @@ static inline int z_vrfy_device_init(const struct device *dev)
 
 extern void boot_banner(void);
 
+#ifdef CONFIG_BOOTARGS
+extern const char *get_bootargs(void);
+static char **prepare_main_args(int *argc)
+{
+	const char *bootargs = get_bootargs();
+	static char *argv_buf[CONFIG_BOOTARGS_ARGV_BUFFER_SIZE] = { NULL };
+	static char argv_strings_buf[CONFIG_BOOTARGS_ARGS_BUFFER_SIZE];
```

Comment:
Nitpick, but with only a little extra code you can put the argv pointers at the end of the buffer while you copy from the start, and avoid having to configure the two separately.

Reply:
I believe this is fixed as well